### PR TITLE
[FIRParser] Parse type alias

### PR DESCRIPTION
**Pull request by uenoku**
This adds FIR parser support for type alias. Type alias for non-base types will be inlined. 

```
circuit Top:
  type Complex = { real: SInt<10>, imag: SInt<10> }
  type ProbeComplex = Probe<Complex>
  module Top:
    output out: ProbeComplex
```

```
 firrtl.module @Top(out %out: !firrtl.probe<alias<Complex, bundle<real: sint<10>, imag: sint<10>>>>) {
 ...

```

### CODE REVIEW

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 1]
```diff
@@ -3764,6 +3780,24 @@ ParseResult FIRCircuitParser::parseModule(CircuitOp circuit,
   return success();
 }
 
+// Parse a type declaration.
+ParseResult FIRCircuitParser::parseTypeDecl() {
+  StringRef id;
+  FIRRTLType type;
+  consumeToken();
+  auto loc = getToken().getLoc();
+  if (parseId(id, "expected wire name") ||
```
Comment (dtzSiFive) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:3789]:
```suggestion
  if (parseId(id, "expected alias name") ||
```

**Review by dtzSiFive — COMMENTED** [review]
(see other PR, we have some modelling concerns to sort out.  As-is I think these would end up in the open aggregates and `LowerOpenAggs` is ref-only for now, for better or for worse)

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp (hunk 1 above)
Comment (uenoku) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp]: Thanks!

**Comment by uenoku** [comment]
Since we only allow base types internally, alias for ref types will be inlined and removed.

**Comment by uenoku** [comment]
This PR will be subsumed by #5167

**Pull request closed without merging**